provision/trigger.go

```diff
@@ -0,0 +1,76 @@
+package provision
+
+import (
+	"fmt"
+	"scaling_manager/cluster"
+)
+
+// Input:
+// Description:
+//
+//	getRecommendation will fetch the recommendation from recommendation queue and clear the queue.
+//	It will populate the command queue which contains all the details to scale out the cluster.
+//
+// Return:
+//
+//	getRecommendation will return the command struct which will contains all the details to scale out the cluster.
+func getRecommendation() Command {
+	var command Command
+	// Fetch the recommendation from the queue
+	// Queue can be stored as document in OS, localy stored data structure or in cache memory.
+	// Clear the recommendation queue
+	return command
+}
+
+// Input:
+// Description:
+//
+//	triggerRecommendation will fetch the recommendation from the queue, get the status of the provisioner
+//	and cluster and trigger the provisioning.
+//
+// Return:
+func triggerRecommendation() {
+	command := getRecommendation()
+	clusterCurrent := cluster.GetClusterCurrent()
+	state := getState()
+	if clusterCurrent.ClusterDynamic.ClusterStatus == "green" ||
+		clusterCurrent.ClusterDynamic.NumRelocatingShards == 0 {
+		if state.CurrentState != "provisioning" && state.CurrentState != "provision" {
```

Comment:
Change this condition according to new state. Because there is a possibility that the state is in provisioned_completed state and at that time we should not provision another recommendation as we are not having a cluster in stable state.